fabfile.py

```diff
@@ -0,0 +1,38 @@
+from fabric.api import run, env, local, put, cd
+
+def deploy():
+    current_revision = local('git rev-parse HEAD', capture=True)
+    create_artifact(current_revision)
+    upload_artifact(current_revision)
+    make_active(current_revision)
+
+def create_artifact(current_revision):
+    """ compress all files into a .tar.gz archive for uploading """
+    archive_path = '/tmp/{revision}.tar.gz'.format(revision=current_revision)
+    local('tar -czf {archive_path} --exclude=.git *'.format(archive_path=archive_path))
+
+def upload_artifact(revision):
+    """ upload the archive to the server and extract it """
+    remote_versions_path = '/home/deploy/versions'
```

Comment:
Please move the shared path-prefixes (e.g. `/home/deploy`) into global constants to ease maintenance.